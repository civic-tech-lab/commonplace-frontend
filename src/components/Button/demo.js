// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import FaCommentingO from 'react-icons/lib/fa/commenting-o';
import FaEnvelopeO from 'react-icons/lib/fa/envelope-o';
import FaExclamationTriangle from 'react-icons/lib/fa/exclamation-triangle';
import FaHeartO from 'react-icons/lib/fa/heart-o';
import FaShareSquareO from 'react-icons/lib/fa/share-square-o';

import Button from '.';

storiesOf('Button', module).add('assortment', () => (
  <div>
    <Button>Regular button</Button>
    <Button primary>Primary button</Button>
    <Button block>Full width button</Button>
    <Button icon={<FaCommentingO />} primary>
      Reply
    </Button>
    <Button icon={<FaEnvelopeO />}>Message</Button>
    <Button icon={<FaShareSquareO />}>Share</Button>
    <Button icon={<FaHeartO />}>Thank</Button>
    <Button icon={<FaExclamationTriangle />}>Flag</Button>
  </div>
));
