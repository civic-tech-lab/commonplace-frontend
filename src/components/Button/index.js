// @flow
import React, { type Element, type Node } from 'react';
import styled from 'styled-components';

import {
  brandColor,
  textColor,
  font,
  transition,
} from '../../styles/variables';

type Props = {|
  block?: boolean,
  children: Node,
  primary?: boolean,
  icon?: Element<*>,
|};

const HOVER_OPACITY = 0.5;
const ICON_SPACING = '0.5em';
const PADDING_HORIZONTAL = 10;
const PADDING_VERTICAL = 7;

const Root = styled.button`
  background: none;
  border: none;
  color: ${props => (props.primary ? brandColor.primary : textColor.subdued)};
  cursor: pointer;
  display: ${props => (props.block ? 'block' : 'inline-block')};
  font-size: ${font.size.medium}px;
  font-family: ${font.family.ui};
  font-weight: ${font.weight.bold};
  padding: ${PADDING_VERTICAL}px ${PADDING_HORIZONTAL}px;
  outline: none;
  transition: opacity ${transition};
  user-select: none;
  width: ${props => (props.block ? '100%' : 'auto')};

  :hover {
    opacity: ${HOVER_OPACITY};
  }
`;

const FlexWrapper = styled.div`
  align-items: bottom;
  display: flex;
  flex-direction: row;
  justify-content: center;
  width: ${props => (props.block ? '100%' : 'auto')};
`;

const Spacer = styled.span`width: ${ICON_SPACING};`;

const Button = ({ block = false, children, icon, primary }: Props) => (
  <Root block={block} primary={primary}>
    <FlexWrapper block={block}>
      {icon != null &&
        React.cloneElement(icon, { style: { height: '1em', width: '1em' } })}
      {icon != null && <Spacer key="spacer" />}
      {children}
    </FlexWrapper>
  </Root>
);

export default Button;
