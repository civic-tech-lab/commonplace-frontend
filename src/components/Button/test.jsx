// @flow
/* eslint-env jest */
import FaCommentingO from 'react-icons/lib/fa/commenting-o';
import FaEnvelopeO from 'react-icons/lib/fa/envelope-o';
import FaExclamationTriangle from 'react-icons/lib/fa/exclamation-triangle';
import FaHeartO from 'react-icons/lib/fa/heart-o';
import FaShareSquareO from 'react-icons/lib/fa/share-square-o';
import React from 'react';
import renderer from 'react-test-renderer';

import Button from '.';

describe('Button', () => {
  test('renders expected output', () => {
    const tree = renderer
      .create(
        <div>
          <Button>Regular button</Button>
          <Button primary>Primary button</Button>
          <Button block>Full width button</Button>
          <Button icon={<FaCommentingO />} primary>
            Reply
          </Button>
          <Button icon={<FaEnvelopeO />}>Message</Button>
          <Button icon={<FaShareSquareO />}>Share</Button>
          <Button icon={<FaHeartO />}>Thank</Button>
          <Button icon={<FaExclamationTriangle />}>Flag</Button>
        </div>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
