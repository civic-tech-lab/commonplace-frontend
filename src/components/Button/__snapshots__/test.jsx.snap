// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders expected output 1`] = `
<div>
  <button
    className="sc-bdVaJa jGgxWE"
  >
    <div
      className="sc-bwzfXH gzNbGe"
    >
      Regular button
    </div>
  </button>
  <button
    className="sc-bdVaJa loEeYq"
  >
    <div
      className="sc-bwzfXH gzNbGe"
    >
      Primary button
    </div>
  </button>
  <button
    className="sc-bdVaJa lenPuQ"
  >
    <div
      className="sc-bwzfXH fyEczS"
    >
      Full width button
    </div>
  </button>
  <button
    className="sc-bdVaJa loEeYq"
  >
    <div
      className="sc-bwzfXH gzNbGe"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m14.3 20q0 1.2-0.9 2t-2 0.9-2-0.9-0.8-2 0.8-2 2-0.9 2 0.9 0.9 2z m8.6 0q0 1.2-0.9 2t-2 0.9-2-0.9-0.9-2 0.9-2 2-0.9 2 0.9 0.9 2z m8.5 0q0 1.2-0.8 2t-2 0.9-2-0.9-0.9-2 0.9-2 2-0.9 2 0.9 0.8 2z m-11.4-11.4q-4.6 0-8.5 1.5t-6.3 4.2-2.3 5.7q0 2.5 1.6 4.8t4.4 3.9l2 1.1-0.6 2.1q-0.5 2.1-1.6 3.9 3.4-1.4 6.2-3.8l0.9-0.9 1.3 0.2q1.5 0.1 2.9 0.1 4.6 0 8.5-1.5t6.3-4.2 2.3-5.7-2.3-5.7-6.3-4.2-8.5-1.5z m20 11.4q0 3.9-2.7 7.2t-7.3 5.2-10 1.9q-1.6 0-3.2-0.2-4.5 3.9-10.3 5.4-1.1 0.3-2.6 0.5h-0.1q-0.3 0-0.6-0.2t-0.3-0.6v-0.1q-0.1-0.1 0-0.2t0-0.3 0.1-0.2l0.1-0.2 0.2-0.2 0.2-0.2q0.1-0.1 0.7-0.7t0.7-0.9 0.7-0.9 0.8-1.1 0.6-1.3 0.5-1.7q-3.5-2-5.5-4.9t-2-6.3q0-2.9 1.6-5.5t4.2-4.6 6.4-3.1 7.8-1.1 7.8 1.1 6.3 3.1 4.3 4.6 1.6 5.5z"
          />
        </g>
      </svg>
      <span
        className="sc-htpNat dbBidq"
      />
      Reply
    </div>
  </button>
  <button
    className="sc-bdVaJa jGgxWE"
  >
    <div
      className="sc-bwzfXH gzNbGe"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m37.1 33.6v-17.2q-0.7 0.8-1.5 1.5-6 4.6-9.5 7.5-1.1 1-1.9 1.5t-1.9 1.1-2.3 0.6h0q-1.1 0-2.3-0.6t-1.9-1.1-1.9-1.5q-3.5-2.9-9.5-7.5-0.8-0.7-1.5-1.5v17.2q0 0.3 0.2 0.5t0.5 0.2h32.8q0.3 0 0.5-0.2t0.2-0.5z m0-23.5v-0.5l0-0.3 0-0.3-0.2-0.2-0.2-0.2-0.3 0h-32.8q-0.3 0-0.5 0.2t-0.2 0.5q0 3.7 3.2 6.3 4.3 3.4 9 7.1 0.1 0.1 0.8 0.7t1 0.8 1 0.7 1.1 0.6 1 0.2h0q0.5 0 1-0.2t1.1-0.6 1-0.7 1-0.8 0.8-0.7q4.7-3.7 9-7.1 1.2-0.9 2.2-2.6t1-2.9z m2.9-0.8v24.3q0 1.4-1 2.5t-2.6 1h-32.8q-1.5 0-2.6-1t-1-2.5v-24.3q0-1.5 1.1-2.5t2.5-1.1h32.8q1.5 0 2.6 1.1t1 2.5z"
          />
        </g>
      </svg>
      <span
        className="sc-htpNat dbBidq"
      />
      Message
    </div>
  </button>
  <button
    className="sc-bdVaJa jGgxWE"
  >
    <div
      className="sc-bwzfXH gzNbGe"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m32.9 22.1v5.8q0 2.6-1.9 4.5t-4.5 1.9h-18.6q-2.6 0-4.5-1.9t-1.9-4.5v-18.6q0-2.7 1.9-4.6t4.5-1.8h5.7q0.3 0 0.5 0.2t0.2 0.5q0 0.6-0.5 0.7-1.8 0.6-3 1.3-0.2 0.1-0.4 0.1h-2.5q-1.4 0-2.5 1.1t-1 2.5v18.6q0 1.4 1 2.5t2.5 1h18.6q1.5 0 2.5-1t1.1-2.5v-4.8q0-0.4 0.4-0.7 0.6-0.3 1.2-0.8 0.3-0.4 0.8-0.2 0.4 0.2 0.4 0.7z m5.3-11.1l-8.6 8.6q-0.4 0.4-1 0.4-0.2 0-0.5-0.1-0.9-0.4-0.9-1.3v-4.3h-3.6q-7.2 0-9.7 2.9-2.7 3.1-1.7 10.6 0.1 0.5-0.4 0.7-0.2 0.1-0.3 0.1-0.4 0-0.6-0.3-0.2-0.3-0.4-0.7t-0.9-1.5-1.1-2.3-0.9-2.5-0.4-2.7q0-1.1 0.1-2.1t0.3-2 0.6-1.9 1.1-1.9 1.5-1.6 2.1-1.4 2.8-1.1 3.6-0.6 4.3-0.3h3.6v-4.3q0-0.9 0.9-1.3 0.3-0.1 0.5-0.1 0.6 0 1 0.4l8.6 8.6q0.4 0.4 0.4 1t-0.4 1z"
          />
        </g>
      </svg>
      <span
        className="sc-htpNat dbBidq"
      />
      Share
    </div>
  </button>
  <button
    className="sc-bdVaJa jGgxWE"
  >
    <div
      className="sc-bwzfXH gzNbGe"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m37.1 13.3q0-1.8-0.4-3.2t-1.3-2.2-1.8-1.3-2.1-0.7-2.2-0.2-2.5 0.6-2.4 1.4-2 1.6-1.3 1.4q-0.4 0.5-1.1 0.5t-1.1-0.5q-0.5-0.6-1.3-1.4t-2-1.6-2.4-1.4-2.5-0.6-2.2 0.2-2.1 0.7-1.8 1.3-1.3 2.2-0.4 3.2q0 3.8 4.1 7.9l13 12.5 12.9-12.4q4.2-4.2 4.2-8z m2.9 0q0 4.9-5.1 10l-13.9 13.4q-0.4 0.4-1 0.4t-1-0.4l-13.9-13.4q-0.2-0.2-0.6-0.6t-1.3-1.4-1.5-2.2-1.2-2.7-0.5-3.1q0-4.9 2.8-7.7t7.9-2.7q1.4 0 2.8 0.4t2.7 1.3 2.1 1.6 1.7 1.5q0.8-0.8 1.7-1.5t2.1-1.6 2.7-1.3 2.8-0.4q5 0 7.9 2.7t2.8 7.7z"
          />
        </g>
      </svg>
      <span
        className="sc-htpNat dbBidq"
      />
      Thank
    </div>
  </button>
  <button
    className="sc-bdVaJa jGgxWE"
  >
    <div
      className="sc-bwzfXH gzNbGe"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m22.9 30.7v-4.2q0-0.4-0.3-0.6t-0.5-0.2h-4.2q-0.3 0-0.5 0.2t-0.3 0.6v4.2q0 0.3 0.3 0.5t0.5 0.2h4.2q0.3 0 0.5-0.2t0.3-0.5z m-0.1-8.4l0.4-10.2q0-0.3-0.2-0.4-0.3-0.3-0.5-0.3h-5q-0.2 0-0.5 0.3-0.2 0.1-0.2 0.4l0.4 10.2q0 0.3 0.2 0.4t0.5 0.2h4.2q0.3 0 0.5-0.2t0.2-0.4z m-0.3-20.8l17.1 31.4q0.8 1.4 0 2.8-0.4 0.7-1 1.1t-1.5 0.3h-34.2q-0.8 0-1.5-0.3t-1-1.1q-0.8-1.4 0-2.8l17.1-31.4q0.4-0.7 1.1-1.1t1.4-0.4 1.4 0.4 1.1 1.1z"
          />
        </g>
      </svg>
      <span
        className="sc-htpNat dbBidq"
      />
      Flag
    </div>
  </button>
</div>
`;
